### ISSUE TITLE
ConjureEvalCurrentForm fails in presence of Clojure metadata reader

### ISSUE THREAD

**Issue by philomates** [Olical/conjure#444]
hi there, I recently noticed troubles trying to evaluate clojure forms with metadata reader data using conjure:

```clojure
^{:some-meta true} 
{:a 1} ;; <-- no result when I set cursor here and call :ConjureEvalCurrentForm

{:a 1} ;; <-- works fine I set cursor here and call :ConjureEvalCurrentForm

;; note that :ConjureEvalRootForm works fine
```

I tried to reproduce the issue in test and wasn't able to, the test I expected to fail passes!:

```fennel
(module conjure.extract-test ...)

(deftest metadata-before-form
  (buffer.with-buf
    ["(ns foo)"
     ""
     "{:1 2} (inc 2)"
     "^{:boom true} {:1 2} (inc 2)"]
    (fn [at]
      (at [3 5])
      (t.pr= {:range {:start [3 0]
                      :end [3 5]}
              :content "{:1 2}"}
             (extract.form {})
             "extract a form that doesn't have metadata")

      ;; expected this one to fail but it succeeds:
      (at [4 19])
      (t.pr= {:range {:start [4 14]
                      :end [4 19]}
              :content "{:1 2}"}
             (extract.form {})
             "extract a form with metadata"))))
```

Have others noticed this issue and/or have any idea what might be responsible?

Thanks!

**Comment by Olical** [comment]
Possibly to do with tree sitter or lack thereof?

**Comment by philomates** [comment]
oh, didn't realize it was used for form extraction when present.
Yes, disabling it resolves the issue. So shall I close this issue and look to open one on the tree-sitter-clojure repo? Or maybe it already exists with https://github.com/sogaiu/tree-sitter-clojure/issues/21 ?

**Comment by Olical** [comment]
Pushed a change to develop that allows the tree sitter one to take `^{...}` meta syntax into account with the eval. It won't work with all prefixes like `^String` etc right now because that opens the floodgates for some other issues, but I think this is an improvement.

So your specific example works with tree sitter now for me. As does `@foo` derefing, part of the same solution. Solved it for a colleague so I thought I'd expand the config to your issue too.

**Comment by Olical** [comment]
Okay, ignore the `^String`, that's a different syntax and would never work! But I've made `^{` and `^:` prefixes work, so that's neat!

**Comment by NoahTheDuke** [comment]
It looks like this doesn't work with certain `ns` forms. I found this through `clojure.test`, I can't eval the buffer or the namespace form while it has the `^{:author "..." :doc "..."}` form. If I delete the metadata, then it can be evaled.

I was able to narrow the failing case down to this form:

```clojure
(ns ^{:doc "







"} cool-ns)
```

It seems that it fails after 8 newlines in a metadata docstring, regardless of the other contents.

**Comment by Olical** [comment]
Good catch, thank you! I'll get a test added and fix this as quickly as
possible. I thought I'd already dealt with this a long time ago, but maybe
there's something different about this one I never accounted for.

What makes it a liiiitle tricky to fix quickly is that we're deep in
refactor land on the nfnl-migration branch which will conflict heavily with
any main line branch changes. So I might have to write a patch for both
branches or we just fix it on the migration branch and move you over there
for a little bit while we finish it off.

I'll have a think! (we're kind of modernising all of the Fennel code in the
Conjure codebase to make maintenance a lot easier, so it's a rough time for
bugfixes because they have to be done in two fairly different branches)

**Comment by NoahTheDuke** [comment]
There's no rush, I ran into this 2 years ago and keep forgetting to open a new issue about it lmao.
